### PR TITLE
update AIP-31: specs of detached delegators allowlist

### PR DESCRIPTION
**Pull request by alexfilip2**

### CODE REVIEW

File: aips/aip-31.md [hunk 1]
```diff
@@ -37,6 +45,10 @@ Some delegation pools may require a KYC process before accepting delegated stake
 
 Delegation pools with allowlisting will have to manage their own KYC process. 
 
+The minimum stake requirement for active and pending-inactive stake per delegator is 10 APT.
+A delegator owning exactly 10 APT active stake might not be able to unlock or be evicted by the pool owner: 10 APT could be worth 9,99..9 APT when unlocked due to rounding errors caused by the "active to pending-inactive" conversion and the unlock would revert.\
```
Comment (junkil-park) [inline aips/aip-31.md:49]: Remove "\" at the end of the sentence?

File: aips/aip-31.md [hunk 2]
```diff
@@ -37,6 +45,10 @@ Some delegation pools may require a KYC process before accepting delegated stake
...
+As rewards are added on this delegator's active stake, the active balance becomes > 10 APT which unblocks them. However, in case the validator is inactive and the delegator is not allowlisted, they don't produce rewards and also cannot add stake to get unblocked.
```
Comment (michelle-aptos) [inline aips/aip-31.md:50]: how will a delegator know if they've been evicted from the allowlist?

File: aips/aip-31.md [hunk 3]
```diff
@@ -20,7 +20,15 @@ This AIP proposes to add a flag in `delegation_pool.move` to allow “permission
 
 Some delegation pools may require a KYC process before accepting delegated stake for compliance reasons. Since delegation pools are permissionless, this feature gives the pool owner more control over who the delegation pool can accept stake from. 
 
-## Rationale
+## Specification
+
+Any account can construct their own allowlist of addresses, independent of the delegation pool, by interacting with the `aptos_framework::delegation_pool_allowlist` module.
```
Comment (michelle-aptos) [inline aips/aip-31.md:25]: is there a way to check if allowlisting is enabled?

File: aips/aip-31.md (hunk 2 above)
Comment (alexfilip2) [reply aips/aip-31.md:50]: Individual evicts are not explicitly stored on-chain, they are only logged as events. A delegator in a specific pool finds out that they've been evicted by checking that they are not part of the allowlist and their entire active stake have been unlocked, but not by themselves.

File: aips/aip-31.md (hunk 3 above)
Comment (alexfilip2) [reply aips/aip-31.md:25]: Given a delegation pool of address `pool_address`, one can check if allowlisting is enabled by:

1. checking if `delegation_pool::ownership_lookup_enabled(pool_address)` is true

2. checking if `delegation_pool_allowlist::allowlisting_enabled(delegation_pool::get_pool_owner(pool_address))` is true

`delegation_pool` exposes `delegator_allowlisted` which already checks whether ownership lookup is enabled + accesses the allowlist defined on the pool owner.
We can also expose `allowlisting_enabled` on `delegation_pool` to merge steps 1 and 2 from above.

File: aips/aip-31.md (hunk 1 above)
Comment (alexfilip2) [reply aips/aip-31.md:49]: this generates a newline in generated md, is not visibile in md preview

File: aips/aip-31.md [hunk 4]
```diff
@@ -20,7 +20,15 @@ This AIP proposes to add a flag in `delegation_pool.move` to allow “permission
...
+However, a delegation pool will only use the allowlist defined under its owner account. If there is no allowlist created by the owner, the delegation pool is still permissionless.
+
+Accessing the owner address directly from the delegation pool is not possible, therefore a new feature has been introduced to save the owner address within the delegation pool. This can be enabled publicly by calling `aptos_framework::delegation_pool::enable_ownership_lookup` with arguments: `pool_address` and `owner_address` which will be linked from now on.
+
+A delegation pool whose owner has created an allowlist, but doesn't have direct access to the owner address will remain permissionless.
+New delegation pools will have the ownership lookup feature enabled by default.
```
Comment (junkil-park) [inline aips/aip-31.md:32]: Can we also list and explain a bit about the major APIs for operating allowlists (e.g., enable, disable, add, remove)?

**Review by davidiw — COMMENTED** [review]
could we add events containing the pool, action, and affected individual, e.g.,

```
struct AddUser {
  pool: address,
  user: address,
}
```

File: aips/aip-31.md [hunk 5]
```diff
@@ -33,10 +44,15 @@ Some delegation pools may require a KYC process before accepting delegated stake
 5. The pool **owner** can evict an existing delegator that is not part of the allowlist. This operation will move all the delegator's tokens to “pending unstake”. Noted that these tokens will continue to be locked up until the end of the lockup period, and the existing stake will also continue earnings rewards until the end of the lockup period. At the end of the lockup period, funds will be unstaked, but will continue to sit in the pool until the delegator calls withdraw.
 6. Existing pools with the allowlist flag already turned on, can turn it off. Once the allowlist flag is turned off, any staker can add funds to the delegation_pool.
 8. There is no explicit blacklisting function. I.e. (all stakers allowed BUT xxxx address). The reason for this is cause the person behind the blacklisted address can always just create a new account address.
+
```
Comment (junkil-park) [inline aips/aip-31.md:47]: @alexfilip2 , could you add the `## Impact` section?

**Comment by junkil-park** [comment]
This PR migrates to https://github.com/aptos-foundation/AIPs/pull/401.

**Comment by thepomeranian** [comment]
closing this aip per @junkil-park

**Pull request closed without merging**